packages/frontend/src/components/common/TagInput/TagInput.tsx

```diff
@@ -279,9 +283,16 @@ export const TagInput = forwardRef<HTMLInputElement, TagInputProps>(
                 );
             }
 
-            const rejectedTags = tags.filter(
-                (tag) => !validationRegex!.test(tag),
-            );
+            const rejectedTags = tags.filter((tag) => {
+                if (validationFunction) {
+                    return (
+                        !validationRegex!.test(tag) ||
+                        (validationFunction ? !validationFunction(tag) : true)
```

Comment:
I was hesitant about using validation regex in the first place because v7 version does not provide any of that https://mantine.dev/core/tags-input/?t=props
I think the reason behind that is react-hook-form and form validation. we can ship this feature but eventually we have to remove both of the props and provide alternative solution to the problem

Reply:
Agreed. I looked at the v7 api and decided not to try to make it work here. 

I suggest if we upgrade to v7 we use the new input anyway. In any case I think this function is closer to what's there than passing the regex. We could consider removing the regex.

Reply:
> I suggest if we upgrade to v7 we use the new input anyway
💯 absolutely. this is temporary